scripts/build_renewable_profiles.py

```diff
@@ -217,6 +217,36 @@
 GEBCO_CRS = "EPSG:4326"
 
 
+def check_cutout_match(cutout, geodf):
+    x_low_out_cutout = geodf.x.min() < cutout.coords["x"].min()
+    x_up_completly_out_cutout = geodf.x.max() < cutout.coords["x"].min()
+
+    x_up_out_cutout = geodf.x.max() > cutout.coords["x"].max()
+    x_low_completly_out_cutout = geodf.x.min() > cutout.coords["x"].max()
+
+    y_low_out_cutout = geodf.y.min() < cutout.coords["y"].min()
+    y_up_completly_out_cutout = geodf.y.max() < cutout.coords["y"].min()
+
+    y_up_out_cutout = geodf.y.max() > cutout.coords["y"].max()
+    y_low_completly_out_cutout = geodf.y.min() > cutout.coords["y"].max()
+
+    assert not (
+        x_low_completly_out_cutout
+        or x_up_completly_out_cutout
+        or y_low_completly_out_cutout
+        or y_up_completly_out_cutout
+    ), (
+        "The requester region is not covered with the provided weather data.\n\r"
+        "Check please the provided cutout. More details on cutout generation are provided in docs:\n\r"
+        "https://pypsa-earth.readthedocs.io/en/latest/tutorial.html#adjust-the-model-configuration \n\r"
+    )
+
+    if x_low_out_cutout or x_up_out_cutout or y_low_out_cutout or y_up_out_cutout:
+        logger.warning(
+            "Weather data does not fully cover the requester region. It's recommended to check the provided cutout. More details are provided in https://pypsa-earth.readthedocs.io/en/latest/tutorial.html#adjust-the-model-configuration"
+        )
```

Comment:
Nice katia! :D Great work and works :)

As a suggestion, using "from shapely.geometry import box" and then exploiting the property bounds of the cutout e.g.
cutout_boundary = box(*cutout.bounds)

using the geodf.total_bounds you can get the bounds of the dataframe likewise and using box again we could get the polygon in shapely.

Finally, we could check if the boundary of the geodf is within the cutout using something like cutout_boundary.contains(geodf.boundary)

I believe using shapely, the implementation may be more compact, what do you think?

Reply:
@davide-f agree that the implementation with shapely geometry functions looks much cleaner. Many thanks for the hint! :)